scripts/utils/generate_strings.py

```diff
@@ -89,30 +91,114 @@ def generateStrings():
 
                             else:
                                 exit(
-                                    f"The comment of {stringId} must be a string or an array of strings"
+                                    f"The comment of {string_id} must be a string or an array of strings"
                                 )
 
                         if len(value) == 0:
-                            stop(stringId)
+                            stop(string_id)
 
                     else:
-                        stop(stringId)
+                        stop(string_id)
 
-                    stringIds.append(
+                    string_ids.append(
                         {
-                            "enumId": pascalize(f"{category}_{key}"),
-                            "stringId": stringId,
+                            "enum_id": pascalize(f"{category}_{key}"),
+                            "string_id": string_id,
                             "value": value,
                             "comments": comments,
                         }
                     )
 
-        with open(
-            os.path.join(translations_path, "generated", "l18nstrings.h"), "w",
-            encoding="utf-8"
-        ) as output:
-            output.write(
-                """/* This Source Code Form is subject to the terms of the Mozilla Public
+def parseGuideStrings():
+    guides_path = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, "src", "ui", "guides")
+    )
+
+    id_re = re.compile(r"[A-Za-z0-9_]+")
+    guide_ids = []
+
+    for guide_filename in os.listdir(guides_path):
+        if not guide_filename.endswith(".json"): continue
+        with open(os.path.join(guides_path, guide_filename)) as guide_file:
+            guide_json = json.load(guide_file)
+        if not "id" in guide_json:
```

Comment:
Wouldn't make more sense to have all these sanity checks as a linter when adding guide json files ? 

Also, would a JSON schema be better, given how many checks there are (I have never done that personally for my small projects, so no clue how much work it requires).